src/codegen/statements.rs

```diff
@@ -970,33 +966,42 @@ fn destructure(
 /// During a destructure statement, sema only checks if the cast is possible. During codegen, we
 /// perform the cast to a StorageRef if that is the case. The cast function, however, does not add
 /// instructions to the CFG. Here, we add a load instruction to fetch the variables from storage.
-fn destructure_load(
+fn cast_and_try_load(
```

Comment:
The comment doesn't say that a cast also from e.g. `uint32` to `uint64` is done.

Reply:
Fixed